### PR TITLE
Fix midnight logrotation

### LINKED ISSUE #1475: pihole.log is rotated twice on midnight
**In raising this issue, I confirm the following (please check boxes, eg [X]) Failure to fill the template will close your issue:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:**

## 10

---
**[FEATURE REQUEST | QUESTION | OTHER]:**

Please [submit your feature request here](https://discourse.pi-hole.net/c/feature-requests), so it is votable by the community.  It's also easier for us to track.

**[BUG | ISSUE] Expected Behaviour:**
`pihole.log` should only be rotated once every day

**[BUG | ISSUE] Actual Behaviour:**
`pihole.log` is rotated twice every day since having merged #1460

**[BUG | ISSUE] Steps to reproduce:**
Wait until midnight

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**

## 10

---

Fix logrotation: manual flushing should be done twice, but automated rotation at midnight should only be done *once*!

Fixes #1475 

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

**Comment by cybrnook** [comment]
I think this looks better. I like that you have removed the opening condition of:

"if command -v /usr/sbin/logrotate >/dev/null; then"

I was thinking on my way to work, that I don't see how this condition would ever not be met. So I don't think adding an "else" to anything after that is meaningful, since it would never make it that far.

I like your correction of adding the pre "if" statement:

"if [[ "$@" == "once" ]]; then"

I will test your three PR's this evening if you wish. Visually I think it looks okay.

File: advanced/Scripts/piholeLogFlush.sh [hunk 1]
```diff
@@ -8,17 +8,38 @@
...
+    # Logrotate once
+    /usr/sbin/logrotate --force /etc/pihole/logrotate
+  else
+    # Copy pihole.log over to pihole.log.1
+    # and empty out pihole.log
+    # Note that moving the file is not an option, as
+    # dnsmasq would happily continue writing into the
+    # moved file (it will have the same file handler)
+    cp /var/log/pihole.log /var/log/pihole.log.1
+    echo " " > /var/log/pihole.log
```
Comment (Mcat12) [inline advanced/Scripts/piholeLogFlush.sh]: Will this cause a space to be put in front of the first log line, possibly messing up the parsing of the first line? However, I do see that it's the original behavior.

File: advanced/Scripts/piholeLogFlush.sh [hunk 2]
```diff
@@ -8,17 +8,38 @@
...
+  if command -v /usr/sbin/logrotate >/dev/null; then
+    # Logrotate twice to move all data out of sight of FTL
+    /usr/sbin/logrotate --force /etc/pihole/logrotate; sleep 3
+    /usr/sbin/logrotate --force /etc/pihole/logrotate
+  else
+    # Flush both pihole.log and pihole.log.1 (if existing)
+    echo " " > /var/log/pihole.log
+    if [ -f /var/log/pihole.log.1 ]; then
+      echo " " > /var/log/pihole.log.1
+    fi
```
Comment (Mcat12) [inline advanced/Scripts/piholeLogFlush.sh]: See other comment.

Comment (cybrnook) [inline advanced/Scripts/piholeLogFlush.sh]: You think likely better to have "" instead?

Comment (cybrnook) [inline advanced/Scripts/piholeLogFlush.sh]: I guess really it doesn't matter as all it's doing is creating an initial file. So I assume any subsequent entry would be >> and thus show up after the hidden character.

Maybe it's better to do a time stamp or something to  reflect when the log was rotated.

Something like:

echo "Log rotated $(date)" > /var/log/pihole.log

**Comment by DL6ER** [comment]
```
echo " " > /var/log/pihole.log
```
will create an empty file + add a space + add a newline. Whatever `dnsmasq` wants to write will start in the second line and will not have any space in front.

> Maybe it's better to do a time stamp or something to reflect when the log was rotated.
> Something like:
> echo "Log rotated $(date)" > /var/log/pihole.log

I disagree, as this will add a line that should not be there as `dnsmasq` would never write such a line. It may have unforeseeable effects on any parsing software (which does not necessarily have to be ours!).

**Pull request merged**